### PR TITLE
[7323] Added max_eps new configuration for syscollector module

### LINKED ISSUE #7323: Establish EPS configuration for syscollector synchro
## Description
This issue aims to add a new configuration under a synchronization tag, in which the number of messages to send in a period of 1 second is established.

for them, you must look for an approach equal to the one that fim (syscheckd) currently has in its configuration.

https://documentation.wazuh.com/4.0/user-manual/reference/ossec-conf/syscheck.html#synchronization

## Expected default configuration:
```<!-- Database synchronization settings -->
    <synchronization>
      <max_eps>10</max_eps>
    </synchronization>
```

## DoD
- [x] Modify configuration parser / Add unit test
- [x] Add/modify default configuration to 10
- [X] Add integration test / Ask QA team -> Technical debt
- [x] PR CI Pass

### PR DESCRIPTION
**Pull request by tinachoB**
|Related issue|
|---|
|Closes #7323 |

## **Description**
This issue aims to add a new configuration under a synchronization tag, in which the number of messages to send in a period of 1 second is established.

for them, you must look for an approach equal to the one that fim (syscheckd) currently has in its configuration.

https://documentation.wazuh.com/4.0/user-manual/reference/ossec-conf/syscheck.html#synchronization

## **Expected default configuration**
```
    <synchronization>
      <max_eps>10</max_eps>
    </synchronization>
```

## **DoD**
- [X] Implement the e2e configuration flow.
- [X] Add/modify default configuration to 10
- [x] PR CI Pass

### CODE REVIEW

File: src/wazuh_modules/wm_syscollector.c [hunk 1]
```diff
@@ -99,6 +97,8 @@ void* wm_sys_main(wm_sys_t *sys)
     }
     if (syscollector_start_ptr) {
         mtinfo(WM_SYS_LOGTAG, "Starting Syscollector.");
+        syscollector_sync_max_eps = sys->sync.sync_max_eps;
+        mtinfo(WM_SYS_LOGTAG, "MNB: sync_max_eps vale: %ld", syscollector_sync_max_eps);
```
Comment (Dwordcito) [inline src/wazuh_modules/wm_syscollector.c:101]: remember delete this

File: src/config/wmodules_syscollector.c [hunk 2]
```diff
@@ -22,14 +22,35 @@ static const char *XML_PACKAGES = "packages";
...
+
+static void parse_synchronization_section(wm_sys_t * syscollector, XML_NODE node) {
+    const char *XML_DB_SYNC_MAX_EPS = "max_eps";
+    const int XML_DB_SYNC_MAX_EPS_SIZE = 7;
+    for (int i = 0; node[i]; ++i) {
+        if (strncmp(node[i]->element, XML_DB_SYNC_MAX_EPS, XML_DB_SYNC_MAX_EPS_SIZE) == 0) {
+            char * end;
+            const long value = strtol(node[i]->content, &end, 10);
+
+            if (value < 0 || value > 1000000 || *end) {
```
Comment (konangc87) [inline src/config/wmodules_syscollector.c:35]: avoid magic numbers.

**Review by konangc87 — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/wm_syscollector.c (hunk 1 above)
Comment (tinachoB) [reply src/wazuh_modules/wm_syscollector.c:101]: Will do as soon as I validate the complete e2e behavior. Thanks for remind me :)

File: src/config/wmodules_syscollector.c (hunk 2 above)
Comment (tinachoB) [reply src/config/wmodules_syscollector.c:35]: Will do

**Review by konangc87 — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**